src/Core/Managed/Operation.AL.Shared/TelemetryClientExtensions.cs

```diff
@@ -21,47 +21,95 @@ public static class TelemetryClientExtensions
+        /// <returns>Operation item object with a new telemetry item having current start time and timestamp.</returns>
+        public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemetryClient, string operationName, string operationId, string parentOperationId = null) where T : OperationTelemetry, new()
+        {
             if (telemetryClient == null)
             {
                 throw new ArgumentNullException("Telemetry client cannot be null.");
             }
 
-            var operation = new AsyncLocalBasedOperationHolder<T>(telemetryClient, new T());
-            operation.Telemetry.Start();
+            var operationTelemetry = new T();
 
-            // Parent context store is assigned to operation that is used to restore call context.
-            operation.ParentContext = AsyncLocalHelpers.GetCurrentOperationContext();
+            if (string.IsNullOrEmpty(operationTelemetry.Name) && !string.IsNullOrEmpty(operationName))
+            {
+                operationTelemetry.Name = operationName;
+            }
 
-            if (string.IsNullOrEmpty(operation.Telemetry.Name) && !string.IsNullOrEmpty(operationName))
+            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Id) && !string.IsNullOrEmpty(operationId))
             {
-                operation.Telemetry.Name = operationName;
+                operationTelemetry.Context.Operation.Id = operationId;
             }
 
-            telemetryClient.Initialize(operation.Telemetry);
+            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.ParentId) && !string.IsNullOrEmpty(parentOperationId))
+            {
+                operationTelemetry.Context.Operation.ParentId = parentOperationId;
+            }
+
+            return StartOperation(telemetryClient, operationTelemetry);
+        }
+
+        /// <summary>
+        /// Creates an operation object with a given telemetry item. 
+        /// </summary>
+        /// <typeparam name="T">Type of the telemetry item.</typeparam>
+        /// <param name="telemetryClient">Telemetry client object.</param>
+        /// <param name="operationTelemetry">Operation to start.</param>
+        /// <returns>Operation item object with a new telemetry item having current start time and timestamp.</returns>
+        public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemetryClient, T operationTelemetry) where T : OperationTelemetry
+        {
+            if (telemetryClient == null)
+            {
+                throw new ArgumentNullException("Telemetry client cannot be null.");
+            }
+
+            if (operationTelemetry == null)
+            {
+                throw new ArgumentNullException("operationTelemetry cannot be null.");
+            }
+
+            var operation = new AsyncLocalBasedOperationHolder<T>(telemetryClient, operationTelemetry);
+            operationTelemetry.Start();
```

Comment:
I agree - ordering is not logical. If calculating what's inside - `Start` should be the very last operation. If measuring total - it should be first. Not a big deal, thanks for noticing. You can switch the order